crates/valence_boss_bar/src/components.rs

```diff
@@ -0,0 +1,76 @@
+use bevy_ecs::prelude::{Component, Entity, Bundle};
+use valence_core::{uuid::UniqueId, text::Text, protocol::{Encode, Decode}};
+use bitfield_struct::bitfield;
+
+#[derive(Bundle)]
+pub struct BossBarBundle {
+    pub id: UniqueId,
+    pub title: BossBarTitle,
+    pub health: BossBarHealth,
+    pub style: BossBarStyle,
+    pub flags: BossBarFlags,
+    pub viewers: BossBarViewers,
+}
+
+impl BossBarBundle {
+
+    pub fn new(title: Text, color: BossBarColor, division: BossBarDivision, flags: BossBarFlags) -> BossBarBundle {
+        BossBarBundle {
+            id: UniqueId::default(),
+            title: BossBarTitle(title),
+            health: BossBarHealth(1.0),
+            style: BossBarStyle {
+                color,
+                division,
+            },
+            flags,
+            viewers: BossBarViewers(Vec::new()),
+        }
+    }
+
+}
+
+#[derive(Component, Clone)]
+pub struct BossBarTitle(pub Text);
+
+#[derive(Component)]
+pub struct BossBarHealth(pub f32);
+
+#[derive(Component)]
+pub struct BossBarStyle {
+    pub color: BossBarColor,
+    pub division: BossBarDivision,
+}
+
+#[derive(Component, Copy, Clone, PartialEq, Eq, Debug, Encode, Decode)]
+pub enum BossBarColor {
+    Pink,
+    Blue,
+    Red,
+    Green,
+    Yellow,
+    Purple,
+    White,
+}
+
+#[derive(Component, Copy, Clone, PartialEq, Eq, Debug, Encode, Decode)]
+pub enum BossBarDivision {
+    NoDivision,
+    SixNotches,
+    TenNotches,
+    TwelveNotches,
+    TwentyNotches,
+}
+
+#[bitfield(u8)]
+#[derive(Component, PartialEq, Eq, Encode, Decode)]
+pub struct BossBarFlags {
+    pub darken_sky: bool,
+    pub dragon_bar: bool,
+    pub create_fog: bool,
+    #[bits(5)]
+    _pad: u8,
+}
+
+#[derive(Component)]
+pub struct BossBarViewers(pub Vec<Entity>);
```

Comment:
I'm somewhat on the fence about having this component, because we're very close to having visibility layers. #362

Reply:
You mean the visibility layers could replace this? If they can then I will try to integrate that once your PR is merged

Reply:
Agreed